Create Blog Post “vonage-cli-now-in-beta” (#1618)
Nexmo/deved-platform: content/blog/en/vonage-cli-now-in-beta.md

---
title: Vonage CLI Now In Beta
description: The Vonage CLI is now in beta - and here is how you can help out!
author: kellyjandrews
published: false
published_at: 2021-08-20T12:59:30.323Z
updated_at: 2021-08-20T12:59:30.361Z
category: announcement
tags:
  - CLI
  - oclif
  - Lerna
comments: true
spotlight: false
redirect: ""
canonical: ""
outdated: false
replacement_url: ""
---

> Throughout a career, specific projects naturally stand out. The [Vonage CLI](https://github.com/Vonage/vonage-cli) is one of those projects for me, and I can now tell you that the CLI is in beta!  The future potential of the CLI and the power of a new plugin architecture has me excited to bring you the news!

Reviewer: Maybe have a place where CLI is defined, just for people that may not know what it stands for.
```suggestion
Throughout a career, specific projects naturally stand out. The [Vonage CLI](https://github.com/Vonage/vonage-cli) is one of those projects for me, and I can now tell you that the CLI (Command Line Interface) is in beta!  The future potential of the CLI and the power of a new plugin architecture has me excited to bring you the news!
```

## How It's Built

> When first talking about the new CLI architecture, plugins, and modularizing our packages better, two essential critical requirements were on the list.  

Reviewer: Are plugins and modularizing our packages better the two things? If they are, maybe something like
```suggestion
When initially planning the new CLI architecture, there were two essential requirements: plugins and modularizing our packages better.  
```

> We chose the [oclif CLI framework](https://oclif.io/) written by the team at Heroku and used in their tooling. Ultimately the decision came down to ease of use, and the plugin architecture built into the framework satisfied both of the main requirements we were after. 

Reviewer: oclif stands for Open CLI Framework, so it's like saying ATM machine. Maybe something like:
```suggestion
We chose [oclif](https://oclif.io/), a CLI framework written by the team at Heroku for internal development. Ultimately, the decision came down to ease of use, and the plugin architecture built into the framework satisfied both of the main requirements we were after. 
```

The plugin architecture will help us to build, deliver, and support additional use cases and functionality that would have been otherwise impossible in our current CLI.

> Along with being able to create and install plugins, quickly versioning and deploying them is equally critical.  These reasons are why I chose [Lerna](https://lerna.js.org/) to support our multi-package repository. Having the ability to link local packages together quickly made the actual process so much easier. 

Reviewer: A lot of quickly's in here. haha
```suggestion
Along with being able to create and install plugins, the processes for quickly versioning and deploying them are equally critical.  These reasons are why I chose [Lerna](https://lerna.js.org/) to support our multi-package repository. Having the ability to link local packages together made versioning and deploying plugins so much easier. 
```

> The real power, I found, however,  was in deployments. The Lerna CLI prompts you through versioning packages and only releasing new versions of any updated software (or keep versions the same if you want).  In all, I fell in love with Lerna and enjoy this methodology. 

Reviewer: ```suggestion
The real power I found was in deployments. The Lerna CLI steps you through the process of versioning packages and offers the option of only releasing new or keeping the same versions of any updated software. In all, I fell in love with Lerna and enjoy this methodology. 
```

## How to Help

> The beta is currently available for testing.  You can quickly install using [`npm` ](https://www.npmjs.com/):

Reviewer: ```suggestion
The beta is currently available for testing and can be installed using [`npm` ](https://www.npmjs.com/):
```

```shell
npm install -g @vonage/cli@beta
```

> Once installed, you can get around using the `--help` flag on any command to learn more about the arguments and flags available. The first thing you will want to do is get your authentication set up. You can do with the following:

Reviewer: Did you want to mention that they can get the API key and secret from the dashboard if they are not familiar, or are you assuming they have already familiarity with Vonage and have an account and already have projects set up?
```suggestion
Once installed, you can use the `--help` flag on any command to learn more about the arguments and flags available. The first thing you will want to do is get your authentication set up with the following:
```

```shell
vonage config:set --apiKey=XXXXXX --apiSecret=XXXXXX
```

> We also have two plugins in beta - In-App Conversations and  Users. Once you have the CLI installed, you can also add these:

Reviewer: Wanted to try and remove an also.
```suggestion
We have two available plugins in beta: In-App Conversations and Users. Once you have the CLI installed, you can also add these:
```

```shell
vonage plugins:install @vonage/cli-plugin-conversations@beta
vonage plugins:install @vonage/cli-plugin-users@beta
```
Both users and conversations will expect you to have an application created:

```shell
vonage apps:create
```

## Feedback 

> I'm working on getting this to v1.0 as quickly as I can, and I won't get it entirely right without your help as well. Install it, break things, and send me some issues on [the Vonage CLI Github repo](https://github.com/Vonage/vonage-cli/issues), ask questions in our [Slack community](https://developer.vonage.com/community/slack) in the `#vonage-cli` channel, or email us at `devrel@vonage.com`. 

Reviewer: Maybe be a little more positive on the outcome of the process. hahaha Maybe something like:
```suggestion
I am actively developing to release a v1.0 of the CLI soon. Feedback is appreciated and crucial to help make this tool as useful to as many developers as possible. Install it, create some applications (your own or follow along with our [tutorials](https://developer.nexmo.com/client-sdk/tutorials/)), and report any issues to [the Vonage CLI Github repo](https://github.com/Vonage/vonage-cli/issues). Feel free to ask questions in our [Slack community](https://developer.vonage.com/community/slack) in the `#vonage-cli` channel or email us at `devrel@vonage.com`. 
```
